app/assets/javascripts/segment_history_graph.coffee

```diff
@@ -15,7 +15,7 @@ $ ->
   hidden_data = []
   first_non_null_attempt = Infinity
 
-  $.each gon.run.splits, (i, split) ->
+  $.each gon.run.raw_splits, (i, split) ->
```

Comment:
I don't understand why it broke, from what I can tell the old `full_segments` code in StatsController produces the same result as the new `segments_with_history` code in CompletedRun, so the swap you did on StatsController line 24 (now 19) shouldn't cause anything that uses `gon.run.raw_splits` to notice a difference. Am I missing something?